src/core/qgssnappingutils.cpp

```diff
@@ -330,6 +330,49 @@ QgsPointLocator::Match QgsSnappingUtils::snapToMap( const QgsPointXY &pointMap,
 }
 
 
+QgsFeatureIds QgsSnappingUtils::findVisibleFeatures( QgsVectorLayer *layer )
+{
+
+  QgsFeatureIds fids;
+
+  if ( layer )
+  {
+    if ( !mEnableSnappingForInvisibleFeature )
```

Comment:
I think this should not be checked here but rather before the call at line 407 in `prepareIndex()`.